specification/sql/resource-manager/Microsoft.Sql/preview/2022-05-01-preview/SynapseLinkWorkspaces.json

```diff
@@ -0,0 +1,159 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2022-05-01-preview",
+    "title": "SqlManagementClient",
+    "description": "The Azure SQL Database management API provides a RESTful set of web APIs that interact with Azure SQL Database services to manage your databases. The API enables users to create, retrieve, update, and delete databases, servers, and other entities."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/linkworkspaces": {
+      "get": {
+        "tags": [
+          "SynapseLinkWorkspaces"
+        ],
+        "description": "Gets all synapselink workspaces for a database.",
+        "operationId": "SynapseLinkWorkspaces_ListByDatabase",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved synapselink workspaces.",
+            "schema": {
+              "$ref": "#/definitions/SynapseLinkWorkspaceListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 404 ResourceNotFound - The requested resource was not found."
```

Comment:
>404 [](http://example.com/codeflow?start=59&length=3)

When would you return 404 for collection GET

Reply:
when we dont find the database or resource. if linkworkspaces is empty then it returns empty collection.

Reply:
Empty collection should be a 200 with empty array